intuita-webview/src/codemodList/TreeView/index2.tsx

```diff
@@ -0,0 +1,235 @@
+			// @TODO
+			value: hash as unknown as CodemodHash,
+		});
+	}, [executionStack]);
+
+	const [progress, { progressBar, stopProgress }] = useProgressBar(
+		onHalt,
+		// @TODO make generic
+		runningRepomodHash as CodemodHash | null,
+	);
+
+	const handleActionButtonClick = useCallback(
+		(action: RunCodemodsCommand) => {
+			// @TODO
+			const actionValue =
+				action.value as unknown as CodemodNodeHashDigest;
+			if (
+				(progress || executionStack.length) &&
+				action.kind === 'webview.codemodList.dryRunCodemod'
+			) {
+				if (executionStack.includes(actionValue)) {
+					return;
+				}
+				setExecutionStack((prev) => [...prev, actionValue]);
+				return;
+			}
+
+			vscode.postMessage(action);
+		},
+		[executionStack, progress],
+	);
+
+	// @TODO move to other component
+	const actionButtons = (
+		node: CodemodNode,
+		actions: RunCodemodsCommand[],
+		doesDisplayShortenedTitle: boolean,
+	) => {
+		if (node.kind !== 'CODEMOD') {
+			return [];
+		}
+
+		return (actions ?? []).map((action) => {
+			const actionValue =
+				action.value as unknown as CodemodNodeHashDigest;
+			return (
+				<Popover
+					trigger={
+						<VSCodeButton
+							key={action.kind}
+							className={s.action}
+							appearance="icon"
+							onClick={(e) => {
+								e.stopPropagation();
+								handleActionButtonClick(action);
+								if (
+									action.kind ===
+										'webview.codemodList.dryRunCodemod' &&
+									node.codemodKind === 'repomod'
+								) {
+									setRunningRepomodHash(node.hashDigest);
+								}
+							}}
+						>
+							{action.kind ===
+								'webview.codemodList.dryRunCodemod' &&
+								executionStack.includes(actionValue) && (
+									<i className="codicon codicon-history mr-2" />
+								)}
+							{doesDisplayShortenedTitle
+								? action.shortenedTitle
+								: action.title}
+						</VSCodeButton>
+					}
+					popoverText={`${
+						action.kind === 'webview.codemodList.dryRunCodemod' &&
+						executionStack.includes(actionValue)
+							? 'Queued:'
+							: ''
+					} ${action.description}`}
+				/>
+			);
+		});
+	};
+
+	const getActionButtons = (
+		node: CodemodNode,
+		doesDisplayShortenedTitle: boolean,
+	) => {
+		if (node.kind !== 'CODEMOD') {
+			return [];
+		}
+
+		if (node.codemodKind === 'repomod' && runningRepomodHash !== null) {
+			return [];
+		}
+
+		if (
+			// @TODO
+			progress?.codemodHash ===
+				(node.hashDigest as unknown as CodemodHash) &&
+			node.codemodKind === 'executeCodemod'
+		) {
+			return [stopProgress];
+		}
+
+		const actions = [
+			{
+				title: '✓ Dry Run',
+				shortenedTitle: '✓',
+				description:
+					'Run this codemod without making change to file system',
+				kind: 'webview.codemodList.dryRunCodemod' as const,
+				value: node.hashDigest as unknown as CodemodHash,
+			},
+		];
+
+		return actionButtons(node, actions, doesDisplayShortenedTitle);
+	};
+
+	return (
+		<div id="codemodDiscoveryView-treeContainer" tabIndex={0}>
```

Comment:
actually we don't need those `div` with `id` and `tabIndex`, the `IntuitaTreeView` handles that